autoload/airline/extensions/dirvish.vim

```diff
@@ -0,0 +1,39 @@
+" MIT Licsense
+" Plugin: https://github.com/justinmk/vim-dirvish
+" vim: et ts=2 sts=2 sw=2
+
+scriptencoding utf-8
+
+if !exists(':Dirvish')
+  finish
+endif
```

Comment:
I think this is better.

```viml
if !get(g, 'g:loaded_dirvish', 0)
 finish
endif
```

Reply:
With this suggestion i got the following error message:
```
Error detected while processing function <SNR>69_on_colorscheme_changed[1]..<SNR>69_init[6]..airline#extensions#load:
line  315:
E117: Unknown function: airline#extensions#dirvish#init
```

Reply:
Sorry, I mistaked.

```viml
if !get(g, 'loaded_dirvish', 0)
 finish
endif
```

How's this?

Reply:
I will try to create the issue with a minimal vimrc.

Reply:
I added some basic echo statements inside the exist guard. Here is the code i used:

```vim
if !get(g:, 'g:loaded_dirvish', 0)

  if exists('g:loaded_dirvish')
    echom 'variable does exist'
    echom 'dirvish value: ' . g:loaded_dirvish
  else
    echom 'variable does not exist'
  endif

  finish
endif
```

Here is the output i recieved:

```
variable does exist
dirvish value: 1
```

Reply:
This seems to have worked for me:

```vim
if !exists('g:loaded_dirvish')
  finish
endif
```

Reply:
That is strange.
I've added such a diff across my current master, and it's working fine.

```diff
diff --git a/autoload/airline/extensions.vim b/autoload/airline/extensions.vim
index 60716b3..594c835 100644
--- a/autoload/airline/extensions.vim
+++ b/autoload/airline/extensions.vim
@@ -445,7 +445,7 @@ function! airline#extensions#load()
     endfor
   endif

-  if exists(':Dirvish')
+  if exists(':Dirvish') && get(g:, 'airline#extensions#dirvish#enabled', 1)
     call airline#extensions#dirvish#init(s:ext)
     call add(s:loaded_ext, 'dirvish')
   endif
diff --git a/autoload/airline/extensions/dirvish.vim b/autoload/airline/extensions/dirvish.vim
index 5ccab97..0907c0b 100644
--- a/autoload/airline/extensions/dirvish.vim
+++ b/autoload/airline/extensions/dirvish.vim
@@ -4,8 +4,8 @@

 scriptencoding utf-8

-if !exists(':Dirvish')
-  finish
+if !get(g:, 'loaded_dirvish', 0)
+ finish
 endif

 let s:spc = g:airline_symbols.space
@@ -22,10 +22,7 @@ function! airline#extensions#dirvish#apply(...) abort
       \ ? '%{airline#extensions#branch#get_head()}'
       \ : ''

-    let w:airline_section_c =
-      \   '%{join(split((execute("lcd"))))}'
-      \ . s:spc
-      \ . get(w:, 'airline_section_c', g:airline_section_c)
+    let w:airline_section_c = '%{b:dirvish._dir}'

     let w:airline_section_x = ''
     let w:airline_section_y = ''
```

Reply:
Seems like I did not fully catch your bugfixes. Your modified version works for me now!

Reply:
Good.
Please apply these changes to #2143 .